kafka_confluent_metric/getmessages_kafka_confluent_metric.py

```diff
@@ -698,12 +734,20 @@ def get_json_size_bytes(json_data):
     return getsizeof(json.dumps(json_data))
 
 
-def make_safe_project_string(project):
+def make_safe_project_string(project, project_mapping_dict):
     """ make a safe project name string """
     # strip underscores
-    project = PIPE.sub('', project)
-    project = PROJECT_ALNUM.sub('-', project)
-    return project
+    # project = PIPE.sub('', project)
+    # project = PROJECT_ALNUM.sub('-', project)
+    project_list = project.split('|')[1:-1]
+    project_dict = dict()
+    for every_project in project_list:
+        if every_project in project_mapping_dict:
+            project_dict[every_project] = {'custom_project': project_mapping_dict[every_project]['custom_project'],
+                                           'custom_system': project_mapping_dict[every_project]['custom_system']}
+        else:
+            project_dict[every_project] = {'custom_project': every_project, 'custom_system': ''}
+    return project_dict
```

Comment:
@dearmore This code support multiple project. If your test data is:  |123|234|345|, when you set up project_map_id:
123::test1,234::test2, the final project result is: test1, test2 and 345. If you don't set up project_map_id, the final project result is: 123, 234 and 345.